libpod/container_internal.go

```diff
@@ -1446,13 +1446,16 @@ func (c *Container) addImageVolumes(ctx context.Context, g *generate.Generator)
 					return err
 				}
 
-				if err = os.Chown(volumePath, int(uid), int(gid)); err != nil {
-					return errors.Wrapf(err, "error chowning directory %q for volume %q in container %q", volumePath, k, c.ID)
+				if err = os.Chown(srcPath, int(uid), int(gid)); err != nil {
+					return errors.Wrapf(err, "error chowning directory %q for volume %q in container %q", srcPath, k, c.ID)
 				}
 			}
 		}
 
 		if _, err := os.Stat(volumePath); os.IsNotExist(err) {
+			if err = os.MkdirAll(volumePath, 0755); err != nil {
```

Comment:
Do we need a chown on this too? Can you test with the mysql image running as non-root like in #844

Reply:
@mheon I am having a hard time getitng podman working in non-root for me. I want to get this in before the release. Could you please pull the patch in and try it out?

Reply:
Fixed. @mheon PTAL